### PR TITLE
[feat]: [CDS-82040]: Added a new topic processing gitx push events via queue service

### PR DESCRIPTION
**Pull request by adivishy1**
## Describe your changes
Added a new topic for processing the gitx push events when the webhook processing uses the queue service instead of using the mongo db. Locally have tested out the gitx webhook event processing via the queue service and the mongo, works fine.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54319)

### CODE REVIEW

**Comment by sahilhindwani-harness** [comment]
*[`120-ng-manager/config.yml` line 579 at r1](https://reviewable.io/reviews/harness/harness-core/54319#-Nh59yLi8xIcOfyQl76J:-Nh59yLjAGm5M4RzfCAB:bt55lqr) ([raw file](https://github.com/harness/harness-core/blob/09f72684de354744cb8c94b2cbf783f8ef5d13a7/120-ng-manager/config.yml#L579)):*
> ```YAML
> useQueueServiceForGitXWebhook: ${USE_QUEUE_SERVICE_FOR_GITX_WEBHOOK:-false}
> 
> webhookGitXPushEventQueueConfig:
> ```

Can this be different from existing queueConfig for webhook? If not, then we should not add this configuration but rather use the older one.

**Comment by sahilhindwani-harness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java` line 138 at r1](https://reviewable.io/reviews/harness/harness-core/54319#-Nh5ANhL1DIRLwq8KSY1:-Nh5ANhL1DIRLwq8KSY2:biezpqt) ([raw file](https://github.com/harness/harness-core/blob/09f72684de354744cb8c94b2cbf783f8ef5d13a7/120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java#L138)):*
> ```Java
>             .subTopic(webhookDTO.getAccountId())
>             .producerName(moduleName + GITX_WEBHOOK_PUSH_EVENT)
>             .payload(RecastOrchestrationUtils.toJson(webhookDTO))
> ```

nit: Can be simplified by only have producer name in the if condition.

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54319)* status: 7 file(s) to review. 1 change requests. 0 of 2 approvals obtained.

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java [hunk 1]
```diff
@@ -125,12 +128,21 @@ private void enqueueWebhookEvents(WebhookDTO webhookDTO, String topic, String mo
...
+        return EnqueueRequest.builder()
+            .topic(moduleName + WEBHOOK_PUSH_EVENT)
+            .subTopic(webhookDTO.getAccountId())
+            .producerName(moduleName + WEBHOOK_PUSH_EVENT)
+            .payload(RecastOrchestrationUtils.toJson(webhookDTO))
+            .build();
+      }
     }
     // Consumer for branch hook events stream: WebhookBranchHookEventQueueProcessor (in NG manager)
     else if (CREATE_BRANCH == webhookDTO.getGitDetails().getEvent()
```
Comment (sribalij) [inline 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java:148]: Confirming gitSyncSdkService.isGitSimplificationEnabled will not be for CREATE_BRANCH or DELETE_BRANCH?

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/listener/WebhookGitXPushEventQueueProcessor.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    try {
+      log.info("Started processing webhook gitx push event for item id {}", message.getItemId());
+      WebhookDTO webhookDTO = RecastOrchestrationUtils.fromJson(message.getPayload(), WebhookDTO.class);
+      try (NgTriggerAutoLogContext ignore0 = new NgTriggerAutoLogContext("eventId", webhookDTO.getEventId(),
+               webhookDTO.getAccountId(), AutoLogContext.OverrideBehavior.OVERRIDE_ERROR)) {
+        gitXWebhookEventService.processEvent(webhookDTO);
+        return HsqsProcessMessageResponse.builder().success(true).accountId(webhookDTO.getAccountId()).build();
+      }
+    } catch (Exception e) {
+      throw new InternalServerErrorException("Exception while processing webhook gitx push event", e);
```
Comment (sribalij) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/listener/WebhookGitXPushEventQueueProcessor.java:51]: Log error

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java (hunk 1 above)
Comment (adivishy1) [reply 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java:148]: yes, we dont need for the createBranch or the deleteBranch, we need it only for the push event

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/listener/WebhookGitXPushEventQueueProcessor.java (hunk 2 above)
Comment (adivishy1) [reply 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/listener/WebhookGitXPushEventQueueProcessor.java:51]: done

**Review by adivishy1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54319)* status: 7 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/config.yml` line 579 at r1](https://reviewable.io/reviews/harness/harness-core/54319#-Nh59yLi8xIcOfyQl76J:-Nh62pMVACtdbevLQUIN:bnljmw3) ([raw file](https://github.com/harness/harness-core/blob/09f72684de354744cb8c94b2cbf783f8ef5d13a7/120-ng-manager/config.yml#L579)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Can this be different from existing queueConfig for webhook? If not, then we should not add this configuration but rather use the older one.
</blockquote></details>

yes, this is a different queueConfig for webhook
___
*[`120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java` line 138 at r1](https://reviewable.io/reviews/harness/harness-core/54319#-Nh5ANhL1DIRLwq8KSY1:-Nh62dmuDSI_nHoQzDfu:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/09f72684de354744cb8c94b2cbf783f8ef5d13a7/120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/WebhookServiceImpl.java#L138)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

nit: Can be simplified by only have producer name in the if condition.
</blockquote></details>

done

**Comment by sahilhindwani-harness** [comment]
*[`120-ng-manager/config.yml` line 579 at r1](https://reviewable.io/reviews/harness/harness-core/54319#-Nh59yLi8xIcOfyQl76J:-Nh63jRX5akYlEqCjwTb:b-f7dxbw) ([raw file](https://github.com/harness/harness-core/blob/09f72684de354744cb8c94b2cbf783f8ef5d13a7/120-ng-manager/config.yml#L579)):*
<details><summary><i>Previously, adivishy1 (Adithya Viswanathan) wrote…</i></summary><blockquote>

yes, this is a different queueConfig for webhook
</blockquote></details>

but batchSize and threadSleepInMilliseconds are same as existing webhook. Can they be different anytime?

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54319)* status: 7 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

**Comment by adivishy1** [comment]
*[`120-ng-manager/config.yml` line 579 at r1](https://reviewable.io/reviews/harness/harness-core/54319#-Nh59yLi8xIcOfyQl76J:-Nh6FJpz8flFVaJuZek_:b-wnak) ([raw file](https://github.com/harness/harness-core/blob/09f72684de354744cb8c94b2cbf783f8ef5d13a7/120-ng-manager/config.yml#L579)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

but batchSize and threadSleepInMilliseconds are same as existing webhook. Can they be different anytime?
</blockquote></details>

As of now we are not configuring any values, keeping it the same values for all the events, webhookPush, WebhookGitXPush, triggers, webhookBranch, will be fine tuning it later

**Review by sahilhindwani-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54319)* status: 7 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Comment by adivishy1** [comment]
trigger uts

**Comment by adivishy1** [comment]
trigger ut7, ut2

**Comment by adivishy1** [comment]
trigger uts

**Comment by adivishy1** [comment]
trigger ut7

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 2 of 7 files at r1, 2 of 2 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54319)* status: Some of the checks are too old. 4 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/listener/WebhookGitXPushEventQueueProcessor.java` line 52 at r4](https://reviewable.io/reviews/harness/harness-core/54319#-NhQ1Y04CfulKDeNDM7Z:-NhQ1Y04CfulKDeNDM7_:b-swdtkv) ([raw file](https://github.com/harness/harness-core/blob/42331169885e4e6cf357fa1fb63cc7c5600e090c/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/listener/WebhookGitXPushEventQueueProcessor.java#L52)):*
> ```Java
>     } catch (Exception e) {
>       log.error(String.format("Error while processing event item id %s", message.getItemId()), e);
>       throw new InternalServerErrorException("Exception while processing webhook gitx push event", e);
> ```

why do we want to throw exception, cant we just log and continue processing else the thread will get stuck

**Comment by adivishy1** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/listener/WebhookGitXPushEventQueueProcessor.java` line 52 at r4](https://reviewable.io/reviews/harness/harness-core/54319#-NhQ1Y04CfulKDeNDM7Z:-NhQ58v64d07zTLkpz13:b-nmblf) ([raw file](https://github.com/harness/harness-core/blob/42331169885e4e6cf357fa1fb63cc7c5600e090c/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/listener/WebhookGitXPushEventQueueProcessor.java#L52)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

why do we want to throw exception, cant we just log and continue processing else the thread will get stuck
</blockquote></details>

as discussed offline, this exception thrown here is caught and wont lead to a case where the thread will be in a stuck state

**Review by archit-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54319)* status: Some of the checks are too old. 4 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by utkarshchoubey-harness — APPROVED** [review]

**Comment by adivishy1** [comment]
trigger review

**Pull request merged**